src/features/ProfilePage/components/PublicInfo.tsx

```diff
@@ -1,35 +1,88 @@
 import styled from 'styled-components';
-import { useEffect, useState } from 'react';
+import { useState } from 'react';
 import { useTranslation } from 'react-i18next';
 
-import { selectUserInfo } from '@/features/Authentication/userSlice';
+import { selectMentor, selectUser } from '@/features/Authentication/userSlice';
 import { useAppSelector } from '@/store';
-
-import { DEFAULT_ICON_SIZE, palette } from '@/components/constants';
+import { useDebounce } from '@/hooks/useDebounce';
+import { useUpdateMentorMutation } from '@/features/MentorPage/mentorPageApi';
+import { useUpdateUserMutation } from '@/features/Authentication/authenticationApi';
+
+import {
+  DEFAULT_ICON_SIZE,
+  palette,
+  SAVE_DELAY_MS,
+} from '@/components/constants';
 import LabeledInput from '@/components/LabeledInput';
 import Slider from '@/components/Slider';
 import Text from '@/components/Text';
 import TextInput from '@/components/TextInput';
 
+import type {
+  MentorUser,
+  User,
+} from '@/features/Authentication/authenticationApi';
+
 const PublicInfo = () => {
   const { t } = useTranslation('profile');
-  const userInfo = useAppSelector(selectUserInfo);
-
-  const [displayName, setDisplayName] = useState('');
-  const [birthYear, setBirthYear] = useState('');
-  const [area, setArea] = useState('');
-  const [status, setStatus] = useState('');
-  const [isActive, setIsActive] = useState(true);
-  const [story, setStory] = useState('');
+  const user = useAppSelector(selectUser);
+  const mentor = useAppSelector(selectMentor);
+  const [updateUser] = useUpdateUserMutation();
+  const [updateMentor] = useUpdateMentorMutation();
+
+  const [birthYear, setBirthYear] = useState(String(mentor.birth_year));
+  const [displayName, setDisplayName] = useState(mentor.display_name);
+  const [isAbsent, setIsAbsent] = useState(mentor.is_vacationing);
+  const [region, setRegion] = useState(mentor.region);
+  const [status, setStatus] = useState(mentor.status_message);
+  const [story, setStory] = useState(mentor.story);
   const [topicSearchValue, setTopicSearchValue] = useState('');
```

Comment:
Missed this before. But we call these `skills` in every other part of our application and also in the jargon in the whole ylitse-project. it would be nice to use the same name here

Reply:
Good observation! Changed!